MariaDB/server
# MDEV-30953: Add package for Galera server

- [x] *The Jira issue number for this PR is: MDEV-30953*


## Description
Add new package mariadb-server-galera to Debian to recude mariadb-server package dependencies. Galera cluster scripts and libraries are currently in package mariadb-server. Move Galera cluster scripts and libraries to new package mariadb-server-galera as most of the people does not need them and it allows to slim down mariadb-server package

## How can this PR be tested?
As Salsa-CI is currently running out of space most of time (this should be addressed) this could check out:
branch: https://salsa.debian.org/illuusio/mariadb-server/-/pipelines/575821
11.3: https://salsa.debian.org/illuusio/mariadb-server/-/pipelines/575819
or build by hand in Debian sid with `debian/autobake-deb.sh`

This should also go thru MariaDB BB builds

## Basing the PR against the correct MariaDB version
- [x ] *This is a new feature and the PR is based against the latest MariaDB development branch.*


## PR quality check
